src/roktManager.ts

```diff
@@ -25,21 +32,67 @@ export interface IRoktMessage {
     payload: any;
 }
 
+export interface RoktKitFilterSettings {
+    userAttributeFilters?: UserAttributeFilters;
+    filterUserAttributes?: (userAttributes: Dictionary<string>, filterList: number[]) => Dictionary<string>;
+}
+
+export interface IRoktKit  {
+    launcher: IRoktLauncher | null;
+    filters: RoktKitFilterSettings;
+    filteredUser: IMParticleUser | null;
+    userAttributes: Dictionary<string>;
+    selectPlacements: (options: IRoktSelectPlacementsOptions) => Promise<IRoktSelection>;
+}
+
 export default class RoktManager {
     public launcher: IRoktLauncher | null = null;
+    public config: SDKInitConfig | null = null;
+
+    public kit: IRoktKit = null;
+    public filters: RoktKitFilterSettings = {};
+
+    private filteredUser: IMParticleUser | null = null;
     private messageQueue: IRoktMessage[] = [];
 
     constructor() {
         this.launcher = null;
+        this.kit = null;
+        this.filters = {};
+        this.filteredUser = null;
+    }
+
+    public init(config: SDKInitConfig, filteredUser?: IMParticleUser): void {
+        const roktConfig = this.parseConfig(config);
+        const { userAttributeFilters } = roktConfig || {};
+
+        this.filters = {
+            userAttributeFilters,
+            filterUserAttributes: KitFilterHelper.filterUserAttributes,
+        };
+
+        this.filteredUser = filteredUser;
+    }
+
+    public parseConfig(config: SDKInitConfig): IKitConfigs {
+        return config.kitConfigs?.find((roktForwarder: IKitConfigs) => 
+            roktForwarder.name === 'Rokt' && 
+            roktForwarder.moduleId === 181
+        );
     }
 
     public attachLauncher(launcher: IRoktLauncher): void {
-        this.setLauncher(launcher);
+        this.launcher = launcher;
+        this.processMessageQueue();
+    }
```

Comment:
let's remove since now attaching the kit calls processMessageQueue to say that the kit and launcher are ready inside the kit